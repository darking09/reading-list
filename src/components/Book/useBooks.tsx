import { useState, useEffect } from 'react'
export default function useBooks() {
  const [isLoading, setLoading] = useState(true)
  const [classNameString, setClassNameString] = useState('')

  useEffect(() => {
    let classNameStringBase = 'duration-700 ease-in-out group-hover:opacity-75'
    if (isLoading) {
      setClassNameString(`${classNameStringBase} scale-110 blur-2xl grayscale`)
    } else {
      setClassNameString(`${classNameStringBase} scale-100 blur-0 grayscale-0`)
    }
  }, [isLoading])

  return {
    setLoading,
    classNameString
  }
}
