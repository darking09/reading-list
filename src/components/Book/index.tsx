import Image from 'next/image'
import { BookProps } from '@typesFiles/props/book'
import useBooks from './useBooks'

export default function Book({
  book,
}: BookProps) {
  const { setLoading, classNameString } = useBooks(

  )
  return (
    <div className='relative pb-3/2'>
      <Image
        src={book.cover}
        alt={book.title}
        fill
        className={`object-cover rounded-lg ${classNameString}`}
        onLoadingComplete={() => setLoading(false)}
      />
    </div>
  )
}
