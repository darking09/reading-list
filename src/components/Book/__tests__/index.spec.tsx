import Book from '../index'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'


// eslint-disable-next-line react/display-name
jest.mock('next/image', () => ({src, ...props} : any) => <div>{src}</div>)
jest.mock('../useBooks', () => () => ({
  setLoading: jest.fn(),
  classNameString: 'classNameString',
}))

describe('Book', () => {
  it('should render', () => {
    const { container } = render(
      <Book
        book={{
          id: 1,
          pages: 1,
          genre_id: 1,
          author_id: {
            name: 'name',
            otherBooks: [],
          },
          synopsis: 'synopsis',
          year: 1,
          ISBN: 'ISBN',
          is_available: true,
          title: 'title',
          cover: 'cover',
          created_at: new Date(),
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render a book with a cover', () => {
    render(
      <Book
        book={{
          id: 1,
          pages: 1,
          genre_id: 1,
          author_id: {
            name: 'name',
            otherBooks: [],
          },
          synopsis: 'synopsis',
          year: 1,
          ISBN: 'ISBN',
          is_available: true,
          title: 'title',
          cover: 'cover',
          created_at: new Date(),
        }}
      />
    )

    const element = screen.getByText(/cover/)
    expect(element).toBeInTheDocument()
  })
})
