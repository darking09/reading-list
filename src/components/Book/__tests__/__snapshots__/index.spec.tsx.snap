// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render 1`] = `
<div>
  <div
    class="relative pb-3/2"
  >
    <div>
      cover
    </div>
  </div>
</div>
`;
